### PR TITLE
CLC-4991 bCourses style reorganization; change Webcast buttons

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-4991

Amended version of: https://github.com/ets-berkeley-edu/calcentral/pull/3553#issuecomment-90729136

### CODE REVIEW

File: src/assets/stylesheets/_canvas_buttons.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+$bc-canvas-button-primary-hover-border: #525252;
+$bc-canvas-button-hover-background: #e8e8e8;
+$bc-canvas-button-hover-border: #d1d1d1;
+$bc-canvas-off-black: #555;
+$bc-canvas-button-selected-background: #6b6b6b;
+$bc-canvas-button-selected-border: #525252;
+$bc-canvas-button-selected-color: #fff;
+
+// Canvas Button Styles
+.bc-canvas-button, .bc-main .cc-widget-webcast .cc-button {
```
Comment (raydavis) [inline src/assets/stylesheets/_canvas_buttons.scss]: We could override ```cc-button``` styling for _all_ bCourses apps by simply removing ```.cc-widget-webcast``` from this line. However, that would introduce a massive UX change at one go (for example, in Create a Course Site, which currently shows a mix of cc-button and bc-button styles), and we said we'd like to standardize styling one app at a time. This more focused change should be enough to demonstrate the concept.

**Comment by raydavis** [comment]
My proof-of-concept goal was to support both a ngClass approach and a pure SCSS approach for styling those layouts which were needed by both CalCentral and bCourses contexts. This shows the SCSS wrapper approach at the widget level and shows the ngClass approach at the base.html level.

I believe we're in agreement that unshared, bCourses-only templates should just explicitly use "bc-"-specific classes.

**Comment by raydavis** [comment]
As usual, pinging @redconfetti & @christianv .

File: src/assets/stylesheets/_canvas_buttons.scss (hunk 1 above)
Comment (christianv) [inline src/assets/stylesheets/_canvas_buttons.scss]: @raydavis +1

**Comment by christianv** [comment]
front-end lgtm - passing over to @redconfetti for final review.

**Comment by redconfetti** [comment]
Looks good to me.

File: src/assets/javascripts/angular/services/utilService.js [hunk 2]
```diff
@@ -28,6 +28,13 @@
     };
 
     /**
+     * Pass whether this is a bCourses app context or not.
+     */
+    var checkIsBcourses = function() {
+      $rootScope.isBcourses = $route.current.isBcourses;
```
Comment (redconfetti) [inline src/assets/javascripts/angular/services/utilService.js]: @christianv Is it safe now to expect the last statement in a function to be the return with Javascript now, or must this and other functions explicitly use `return` ?

Comment (christianv) [inline src/assets/javascripts/angular/services/utilService.js]: @redconfetti should be an explicit return.

**Pull request merged**